crates/rpc/eth/logs.rs

```diff
@@ -0,0 +1,181 @@
+pub struct LogsRequest {
+    /// The oldest block from which to start
+    /// retrieving logs.
+    /// Will default to `latest` if not provided.
+    pub from_block: BlockIdentifier,
+    /// Up to which block to stop retrieving logs.
+    /// Will default to `latest` if not provided.
+    pub to_block: BlockIdentifier,
+    /// The addresses from where the logs origin from.
+    pub address_filters: Option<AddressFilter>,
+    /// Which topics to filter.
+    pub topics: Vec<TopicFilter>,
+}
+impl RpcHandler for LogsRequest {
+    fn parse(params: &Option<Vec<Value>>) -> Result<LogsRequest, RpcErr> {
+        match params.as_deref() {
+            Some([param]) => {
+                let param = param.as_object().ok_or(RpcErr::BadParams)?;
+                let from_block = {
+                    if let Some(param) = param.get("fromBlock") {
+                        BlockIdentifier::parse(param.clone(), 0)?
+                    } else {
+                        BlockIdentifier::latest()
+                    }
+                };
+                let to_block = {
+                    if let Some(param) = param.get("toBlock") {
+                        BlockIdentifier::parse(param.clone(), 1)?
+                    } else {
+                        BlockIdentifier::latest()
+                    }
+                };
+                let address_filter = param.get("address").and_then(|address| match address {
+                    Value::String(_) | Value::Array(_) => {
+                        Some(serde_json::from_value::<AddressFilter>(address.clone()).unwrap())
+                    }
+                    _ => None,
+                });
+                let topics = param.get("topics").map(|topics| {
+                    serde_json::from_value::<Option<Vec<TopicFilter>>>(topics.clone()).unwrap()
+                });
+                Ok(LogsRequest {
+                    from_block,
+                    address_filters: address_filter,
+                    topics: topics.flatten().unwrap_or_else(Vec::new),
+                    to_block,
+                })
+            }
+            _ => Err(RpcErr::BadParams),
+        }
+    }
+    // TODO: This is longer than it has the right to be, maybe we should refactor it.
+    // The main problem here is the layers of indirection needed
+    // to fetch tx and block data for a log rpc response, some ideas here are:
+    // - The ideal one is to have a key-value store BlockNumber -> Log, where the log also stores
+    //   the block hash, transaction hash, transaction number and its own index.
+    // - Another on is the receipt stores the block hash, transaction hash and block number,
+    //   then we simply could retrieve each log from the receipt and add the info
+    //   needed for the RPCLog struct.
+    fn handle(&self, storage: Store) -> Result<Value, RpcErr> {
+        let Ok(Some(from)) = self.from_block.resolve_block_number(&storage) else {
+            return Err(RpcErr::BadParams);
```

Comment:
This maybe is out of scope of this PR, but instead of a generic BadParams, it would be nice to send an error message that specifies that the issue is on the `from` field

Reply:
Done! Check [this commit](https://github.com/lambdaclass/ethereum_rust/pull/360/commits/4c52daaf086af57701217b6e8e604b9b005531b6), it wasn't too hard to do